src/core/src/service_interfaces/TelemetryWriter.py

```diff
@@ -31,21 +31,22 @@ class TelemetryWriter(object):
     def __init__(self, env_layer, composite_logger, events_folder_path):
         self.env_layer = env_layer
         self.composite_logger = composite_logger
-        self.__is_agent_compatible = False
+        self.__is_telemetry_supported = False
         self.__operation_id = str(datetime.datetime.utcnow())
         self.events_folder_path = None
         self.__telemetry_event_counter = 1  # will be added at the end of each event sent to telemetry to assist in tracing and identifying event/message loss in telemetry
         self.start_time_for_event_count_throttle_check = datetime.datetime.utcnow()
         self.event_count = 1
-        self.agent_env_var_code = Constants.AgentEnvVarStatusCode.AGENT_ENABLED  # code to give details on what went wrong when getting env var
 
         if self.__get_events_folder_path_exists(events_folder_path):
             self.events_folder_path = events_folder_path
-            self.__is_agent_compatible = True
 
-        self.write_event('Started Linux patch core operation.', Constants.TelemetryEventLevel.Informational)
-        self.write_machine_config_info()
-        self.__log_agent_information()
```

Comment:
As discussed, keep this in. telemetrySupported can be added to TelemetryWriter such that it will be set during init
get_path_to_log_files_and_telemetry_dir() in Bootstrapper should be able to extract 'telemetrySupported' from environmentSettings which can be sent back to Bootstrapper, then used in ConfigurationFactory() init to be available as a init var for TelemetryWriter.

This ensures we have Telemetry from as early as possible and and perform telemetry exists check

Reply:
Done!